tests/resources/test_material_run.py

```diff
@@ -42,6 +42,19 @@ def test_register_material_run(collection, session):
     assert "<Material run 'Test MR 123'>" == str(registered)
 
 
+def test_register_all(collection, session):
+    runs = [MaterialRunFactory(name='1'), MaterialRunFactory(name='2'),  MaterialRunFactory(name='3')]
```

Comment:
What if the materials are linked to processes (using gemd-python objects). If you use register_all on the processes first, will the uid info for the processes be updated on the materials?
I basically don't want to run into this problem:
https://citrine.atlassian.net/browse/PLA-3935

Reply:
This tests checks for that situation on `dataset.register_all`:
https://github.com/CitrineInformatics/citrine-python/blob/master/tests/resources/test_dataset.py#L301

Reply:
I did not code in that functionality. It's definitely something I could add, although I'm not sure I like the idea of the client having to manually resolve the differences between objects pre-write and post-write (more than just `uids` change), or of this method having side-effects on its inputs. It would be nice if users could just use the returned list, which is in the same order as its inputs so the association from input to output is easy to make.

Reply:
I just looked at this issue again, and I don't think this is a result of not setting uids client-side. I mean, it is because of the way our normalization/serialization logic works, but I think the underlying cause is that we add and then remove the `auto` uid client-side. Then there's no uid on the process and the material referencing it is serialized, our link swapper just ignores the process instead of complaining that it can't make a link when no uid is present. If we generated ids that we didn't subsequently drop, this wouldn't be a problem.